server/src/server.rs

```diff
@@ -147,6 +147,36 @@ impl CubebDeviceCollectionManager {
     }
 }
 
+struct DeviceMap {
+    devices: Vec<usize>,
+}
+
+impl DeviceMap {
+    fn new() -> DeviceMap {
+        let mut d = DeviceMap {
+            devices: Vec::with_capacity(32)
+        };
+        d.reset();
+        d
+    }
+
+    fn reset(&mut self) {
+        self.devices.clear();
+        // Special case: index 0 always maps to value 0.
+        self.devices.push(0);
+    }
+
+    fn to_handle(&mut self, input: usize) -> usize {
```

Comment:
> I don't think I understand the usage of `DeviceMap`. Is this used to prevent the system info from being leaked to the content process?

Yes, we need a way to validate the cubeb_devid passed to cubeb_stream_init.

Reply:
This is restoring functionality we used to have: https://github.com/djg/audioipc-2/blob/legacy/server/src/lib.rs#L180

That version used a HashSet, but it the current approach is simpler.